hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/debug/DBScanner.java

```diff
@@ -288,6 +309,9 @@ private void processRecords(ManagedRocksIterator iterator,
         batch = new ArrayList<>(batchSize);
         sequenceId++;
       }
+      if ((preFileRecords > 0) && (count >= preFileRecords)) {
+        break;
+      }
```

Comment:
What's the expected behavior when this new `--max-records-per-file` flag is used without `--out`? Right now it looks like the choice that `stdout` is considered "one file" and so this flag overrides the `--length` option:
```
# The DB here has many more than 3 entries
$ ./ozone debug ldb --db=om.db scan --column_family=fileTable -l3 --max-records-per-file=2 | jq '.[].keyName' | wc -l 
2

$ ./ozone debug ldb --db=om.db scan --column_family=fileTable -l2 --max-records-per-file=3 | jq '.[].keyName' | wc -l
2
```

Maybe we should disallow `--max-records-per-file` without `--out`.

Reply:
`-l` is also broken with this new option and I got a bit of a surprise trying to test this 😄 I would have expected 5 files, here not 57 thousand.
```
$ ./ozone debug ldb --db=om.db scan --column_family=fileTable -l10 --max-records-per-file=2 --out=foo
^C
$ ls -l | grep foo | wc -l                                                                           
57343
```

Reply:
Thank you for your comment and review.
I will update soon.

Reply:
> `-l` is also broken with this new option and I got a bit of a surprise trying to test this 😄 I would have expected 5 files, here not 57 thousand.
> 
> ```
> $ ./ozone debug ldb --db=om.db scan --column_family=fileTable -l10 --max-records-per-file=2 --out=foo
> ^C
> $ ls -l | grep foo | wc -l                                                                           
> 57343
> ```

When `--out` is not set, all records are output to stdout.
When `--max-records-per-file` and `-l` exist at the same time, `--max-records-per-file` shall prevail.